src/zulqer/groupPeople/index.ts

```diff
@@ -0,0 +1,62 @@
+interface People {
+  _id: string;
+  name: string;
+  gender: string;
+  email: string;
+  phone: string;
+  city: string;
+}
+
+interface GroupOfPeopleByCity {
+  count: number;
+  city: string;
+  people: Array<People>;
+}
+interface GroupOfPeopleByGender {
+  count: number;
+  gender: string;
+  people: Array<People>;
+}
+
+const groupBy = (arrayOfObject: Array<any>, property: string): Array<any> => {
+  const grouped = [];
+  arrayOfObject.map(function (obj) {
+    if (grouped[obj[property] || property]) {
+      grouped[obj[property] || property].push(obj);
+    } else {
+      grouped[obj[property] || property] = [obj];
```

Comment:
- `grouped` shall be an object, using an array as an object to store at different keys than numeric indices is not a good practice. 
- if the given data `obj` doesn't have the property (say `city`), there is a handling added to assign with the key as property name (`grouped[property]`). Those will be stored at a key with name as `city` which is not good. If the data `obj` doesn't have the property we shall either skip the record or store with a key like `Unassigned`